homeassistant/components/binary_sensor/zha.py

```diff
@@ -87,3 +124,102 @@ def cluster_command(self, tsn, command_id, args):
         elif command_id == 1:
             _LOGGER.debug("Enroll requested")
             self.hass.add_job(self._ias_zone_cluster.enroll_response(0, 0))
+
+
+class Switch(zha.Entity, BinarySensorDevice):
+    """ZHA switch/remote controller/button."""
+
+    _domain = DOMAIN
+
+    class OnOffListener:
+        """Listener for the OnOff ZigBee cluster."""
+
+        def __init__(self, entity):
+            """Initialize OnOffListener."""
+            self._entity = entity
+
+        def cluster_command(self, tsn, command_id, args):
+            """Handle commands received to this cluster."""
+            if command_id in (0x0000, 0x0040):
+                self._entity._set_state(False)
+            elif command_id in (0x0001, 0x0041, 0x0042):
+                self._entity._set_state(True)
+            elif command_id == 0x0002:
+                self._entity._set_state(not self._entity._state)
+
+        def attribute_updated(self, attrid, value):
+            """Handle attribute updates on this cluster."""
+            if attrid == 0:
+                self._entity._state = value
+                self._entity.schedule_update_ha_state()
+
+    class LevelListener:
+        """Listener for the LevelControl ZigBee cluster."""
+
+        def __init__(self, entity):
+            """Initialize LevelListener."""
+            self._entity = entity
+
+        def cluster_command(self, tsn, command_id, args):
+            """Handle commands received to this cluster."""
+            if command_id == 0x0001:  # move_to_level
```

Comment:
move to level command_id should be 0x00.
command_id == 0x01 is "move" command with `[MoveMode(enum8), Rate(uint8)]`